### ISSUE TITLE
Unable to compile yarp_ovrheadset

### ISSUE THREAD

**Issue by GiulioRomualdi** [robotology/yarp#1936]
**Describe the bug**
When I try to compile `yarp_ovrheadset` I get the following error.
~~~
a reference of type "yarp::dev::iBattery::Battery_status&" (not const-qualified) cannot be initialized with a value of type "int"
~~~
**Screenshots**
![schermata del 2018-12-19 16-42-11](https://user-images.githubusercontent.com/16744101/50230523-1c44b080-03ad-11e9-90b1-e6e8cdfe3946.png)

**Configuration:**
 - OS: Windows10
 - yarp version: latest commit in devel https://github.com/robotology/yarp/commit/053d25c23164c7c4a46e056036c6bd17a60ed429
-  compiler: VisualStudio 2017

**Additional context**
The bug can be easily fixed by editing https://github.com/robotology/yarp/blob/053d25c23164c7c4a46e056036c6bd17a60ed429/src/devices/ovrheadset/TextureBattery.cpp#L148
into 
```cpp
yarp::dev::iBattery::Battery_status status;
```

cc @aerydna @kouroshD

**Comment by drdanz** [comment]
@randaz81 I think this might be related to your recent changes to the `Battery` stuff

**Comment by randaz81** [comment]
424d772 is fine, please proceed...

**Comment by Nicogene** [comment]
Fixed by #1943 

Shall we cherry-pick the fix in `master` ?

**Comment by drdanz** [comment]
I think that the changes to the battery are in devel only
